v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/transforms/InitialSplitRangeDoFn.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.jdbc.uniformsplitter.transforms;
+
+import com.google.auto.value.AutoValue;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.uniformsplitter.range.Range;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.Collections;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Generates initial splits for the first partition column, upto max partitions, assuming uniform
+ * density to begin with.
+ */
+@AutoValue
+public abstract class InitialSplitRangeDoFn extends DoFn<Range, ImmutableList<Range>> {
+
+  /**
+   * How many iterations of splits to do. For example if you want 1024 ranges to get created
+   * initially, set this to 10.
+   *
+   * @return split height for initial split.
+   */
+  abstract long splitHeight();
+
+  abstract String tableName();
+
+  private static final Logger logger = LoggerFactory.getLogger(InitialSplitRangeDoFn.class);
+
+  /**
+   * @param input Range indicating Min and Max of the first partition column.
+   * @param out output receiver for a list of initial split of ranges. The ranges are not counted as
+   *     yet.
+   * @param c process context
+   */
+  @ProcessElement
+  public void processElement(
+      @Element Range input, OutputReceiver<ImmutableList<Range>> out, ProcessContext c) {
+
+    logger.info(
+        "RWUPT - Began split process for table {} with initia range as {}", tableName(), input);
```

Comment:
nit: initial range. Also please elaborate on the abbreviation

Reply:
Fixed. Thanks.